docs/guides/es2k/compiling-p4-programs.md

```diff
@@ -1,135 +1,133 @@
 ```
 
 Each P4 program has its own directory and an accompanying README file that
 provides instructions on how to configure the IPU pipeline.
 
-## 4. Compiling a P4 Program
-
-Use `p4c-pna-xxp` to compile a P4 program. We will be using one of the reference
-programs mentioned above.
-
-```bash
-# Set environment variables
-export LD_LIBRARY_PATH=/usr/lib:/usr/lib64:/usr/local/lib:/usr/local/lib64:$LD_LIBRARY_PATH
-export OUTPUT_DIR=/usr/share/mev_reference_p4_files/simple_l3_l4_pna
-
-# Compile p4 program
-p4c-pna-xxp -I/usr/lib -I/usr/share/p4c/p4include -I/usr/share/p4c/idpf-lib \
-            $OUTPUT_DIR/simple_l3_l4_pna.p4 -o $OUTPUT_DIR/simple_l3_l4_pna.s \
-            --p4runtime-files $OUTPUT_DIR/simple_l3_l4_pna.p4info.txt \
-            --context $OUTPUT_DIR/simple_l3_l4_pna.context.json \
-            --bfrt $OUTPUT_DIR/simple_l3_l4_pna.bf-rt.json
-```
-
-The compiler will generate the following files:
-
-- simple_l3_l4_pna.p4info.txt
-- simple_l3_l4_pna.bf-rt.json
-- simple_l3_l4_pna.context.json
-- simple_l3_l4_pna.s
-
-These files are called _P4 artifacts_.
-
-## 5. Generating a Pkg File
-
-Use `cpt` to prepare the P4 artifacts for deployment:
-
-```bash
-cpt --npic --format csr --pbd  -o simple_l3_l4_pna.pkg \
-    cpt_ver.s simple_l3_l4_pna.s
-```
+### Compiling a P4 Program
+
+Use the `p4c` i.e the compiler driver to compile and build packages.
+We will be using one of the reference programs mentioned above as an
+example: `p4-programs/layer-3-forwarding/l3-fwd_p2p``.
+
+The Makefile contains the exact commands to build the artifacts
+for all the reference programs.
+
+1. Copy the entire p4-programs directory from the host to the P4Tools
+   container.
+
+   ```bash
+   sudo docker cp p4-programs/ <p4Tools container id>:/opt/
+   ```
+
+2. Set environment variables in the container.
+
+   ```bash
+   # Check where p4c is installed in the container.
+   [root@a54d354e447e p4-programs] which p4c
+   /opt/p4-tools/p4c/bin/p4c
+   
+   # Set the following env variables in the container prior to building
+   [root@a54d354e447e p4-programs] export P4C_PATH=/opt/p4-tools/p4c/bin
+   export CPT_PATH=/opt/p4-tools/cpt/bin
+   export PATH=$P4C_PATH:$CPT_PATH:$PATH
+   export LD_LIBRARY_PATH=$P4C_PATH/../lib:/usr/local/lib:$LD_LIBRARY_PATH
+   export PREV_ARTIFACT_DIR=
+   ```
+
+3. Compile l3-fwd_p2p,  all artifacts will be generated in artifacts directory.
+
+    ```bash
+   [root@a54d354e447e ~] cd /opt/p4-programs
+   
+   # Compiling the /opt/p4-programs/layer-3-forwarding/l3-fwd_p2p/l3-fwd_p2p.p4
+   # and generating the runtime files and assembly...
+
+   [root@a54d354e447e p4-programs]# make l3-fwd_p2p
+    /opt/p4-tools/p4c/bin/p4c --target idpf --arch pna
+    -I/opt/p4-tools/p4c/bin/../lib -I/opt/p4-tools/p4c/bin/../share/p4c/p4include
+    -I/opt/p4-tools/p4c/bin/../share/p4c/idpf-lib --package-by-domain
+    --p4runtime-files ./artifacts/l3-fwd_p2p/p4Info.txt --save-temps -Xp4c
```

Comment:
Two observations:

1. The quoted string `"--Wdisable ... bf-rt.json"` is the value of the `-Xp4c` parameter. If you're breaking the command line for readability, you generally want the parameter and its argument to be on the same line.
2. If you're breaking a command line for readability, you generally don't want to insert a line break into the middle of a quoted value. This is one case where I consider it acceptable to have more than 80 characters in a line.